File: code-generator/src/main/java/io/github/potjerodekool/openapi/dependency/DependencyChecker.java
package io.github.potjerodekool.openapi.dependency;

import java.util.stream.Stream;

/**
 * A dependency checker to check is a dependency is available.
 */
public interface DependencyChecker {

    /**
     * @param groupId GroupId of the dependency.
     * @param artifactId ArtifactId of the dependency.
     * @return Returns true if the dependency is available during runtime.
     */
    boolean isDependencyPresent(String groupId,
                                String artifactId);

    boolean isClassPresent(String className);

    Stream<Artifact> getProjectArtifacts();
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/dependency/DependencyChecker.kt
package io.github.potjerodekool.openapi.common.dependency

import java.util.stream.Stream

/**
 * A dependency checker to check is a dependency is available.
 */
interface DependencyChecker {
    /**
     * @param groupId GroupId of the dependency.
     * @param artifactId ArtifactId of the dependency.
     * @return Returns true if the dependency is available during runtime.
     */
    fun isDependencyPresent(
        groupId: String?,
        artifactId: String?
    ): Boolean

    fun isClassPresent(className: String?): Boolean

    val projectArtifacts: Stream<Artifact>
}
